impala/tests/test_http_connect.py

```diff
@@ -66,10 +67,85 @@ def __init__(self):
   yield server
 
   # Cleanup after test.
-  if server.httpd is not None:
-    server.httpd.shutdown()
-  if server.http_server_thread is not None:
-    server.http_server_thread.join()
+  shutdown_server(server)
+
+
+@pytest.yield_fixture
+def http_proxy_server():
+  """A fixture that creates a reverse http proxy."""
+
+  class RequestHandlerProxy(SimpleHTTPServer.SimpleHTTPRequestHandler):
+    """A custom http handler that acts as a reverse http proxy. This proxy will forward
+    http messages to Impala, and copy the responses back to the client. In addition, it
+    will save the outgoing http message headers in a class variable so that they can be
+    accessed by test code."""
+
+    # This class variable is used to store the most recently seen outgoing http
+    # message headers.
+    saved_headers=None
+
+    def __init__(self, request, client_address, server):
+      SimpleHTTPServer.SimpleHTTPRequestHandler.__init__(self, request, client_address,
+                                                    server)
+
+    def do_POST(self):
+      # Read the body of the incoming http post message.
+      data_string = self.rfile.read(int(self.headers['Content-Length']))
+      # Save the http headers from the message in a class variable.
+      RequestHandlerProxy.saved_headers = self.decode_raw_headers()
+      # Forward the http post message to Impala and get a response message.
+      response = requests.post(url="http://localhost:28000/cliservice",
```

Comment:
The port should be configurable with env var IMPYLA_TEST_HTTP_PORT

Reply:
Done, thanks